### PR TITLE
General maintenance updates

### PR DESCRIPTION
**Pull request by Nuru**
## what && why
- Update workflow actions to current versions to stop using deprecated features and use current replacements
- Stop trying to update certain package versions (Teleport and `kubectl`) that are far past end-of-life
- Add `kubectl-1.26` (now current/latest version)
- Allow `kubectl` versions 1.20 through 1.23 to fail to update without throwing an error because they are past end-of-life, but keep trying to update them because sometimes urgent security issues get backported anyway.
- Update `k9s` download URL to reflect new asset naming convention as of v0.27.0
- Disable auto-label sync because it removes labels we want to keep, such as manually added labels or labels added on previous commits

### CODE REVIEW

**Review by nitrocode — APPROVED** [review]

File: .github/auto-label.yml [hunk 1]
```diff
@@ -260,6 +260,9 @@ vendor/kubectl-1.24:
 vendor/kubectl-1.25:
 - any: ["vendor/kubectl-1.25/**"]
   all: ["!bin/**", "!tasks/**"]
+vendor/kubectl-1.26:
+- any: ["vendor/kubectl-1.26/**"]
+  all: ["!bin/**", "!tasks/**"]
```
Comment (dudymas) [inline .github/auto-label.yml:265]: wouldn't you want to remove EOL kubectl above?

File: deb/Dockerfile.stable-slim [hunk 2]
```diff
@@ -5,7 +5,7 @@ ENV LC_ALL=C.UTF-8
 ENV PS1="(deb) \w \$ "
 
 RUN apt-get update && \
-    apt-get -y install bash ruby ruby-dev rubygems build-essential curl git zip bzip2 jq
+    apt-get -y -q install bash ruby ruby-dev rubygems build-essential curl git zip bzip2 jq
```
Comment (dudymas) [inline deb/Dockerfile.stable-slim:8]: any reason not to use `-qq`? It also lets you drop the `-y`
```suggestion
    apt-get -qq install bash ruby ruby-dev rubygems build-essential curl git zip bzip2 jq
```

**Review by dudymas — APPROVED** [review]

File: .github/workflows/kubectl-1.13.yml [hunk 3]
```diff
@@ -116,7 +116,7 @@ jobs:
 
       # Publish the artifacts
       - name: "Push artifact to package repository"
-        uses: cloudsmith-io/action@v0.5.1
+        uses: cloudsmith-io/action@v0.5.3
```
[the remark this answers is not in the document]
Comment (Nuru) [reply .github/workflows/kubectl-1.13.yml:119]: These files are automatically generated, and we keep them around so we can regenerate the packages if needed.

File: .github/auto-label.yml (hunk 1 above)
Comment (Nuru) [reply .github/auto-label.yml:265]: This file is automatically generated, and there is no need to remove the configuration for a package. If for some reason something changes, we still want to label the PR appropriately.

File: deb/Dockerfile.stable-slim (hunk 2 above)
Comment (Nuru) [reply deb/Dockerfile.stable-slim:8]: We still want logging in case something goes wrong, and `-qq` suppresses too much.

**Pull request merged**